websurvey/lib/typewheels/responseStore.js

```diff
@@ -12,6 +15,15 @@ class ResponseStore {
   getQa() {
     return this.qa;
   }
+
+  validateFieldValue(field, fieldValue, isRequired) {
+    const res = v.validator(field)(fieldValue);
+    if (!res.valid || (isRequired && fieldValue === " ")) {
+      alert(res.message);
```

Comment:
I would move the responsibility of showing the user something into the UI layer, i.e., into the component itself!